internal/controllers/machineset/machineset_controller.go

```diff
@@ -355,15 +373,17 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 	if machineSet.Spec.MinReadySeconds > 0 &&
 		machineSet.Status.ReadyReplicas == replicas &&
 		machineSet.Status.AvailableReplicas != replicas {
-		return ctrl.Result{RequeueAfter: time.Duration(machineSet.Spec.MinReadySeconds) * time.Second}, nil
+		minReadyResult := ctrl.Result{RequeueAfter: time.Duration(machineSet.Spec.MinReadySeconds) * time.Second}
+		result = util.LowestNonZeroResult(result, minReadyResult)
+		return result, nil
 	}
 
 	// Quickly reconcile until the nodes become Ready.
 	if machineSet.Status.ReadyReplicas != replicas {
-		return ctrl.Result{RequeueAfter: 15 * time.Second}, nil
+		result = util.LowestNonZeroResult(result, ctrl.Result{RequeueAfter: 15 * time.Second})
 	}
 
-	return ctrl.Result{}, nil
+	return result, nil
```

Comment:
```suggestion
	return ctrl.Result{}, nil
```

Reply:
We need to return `result` here or else the value that we calculate in l.350 is lost and never used.

Reply:
Ah thx. Good point